extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -189,13 +194,28 @@ export class AmpStoryDevTools extends AMP.BaseElement {
    * @private
    */
   buildTabs_() {
-    this.tabContents_ = Object.values(DevToolsTab).reduce(
-      (tabContents, tab) => {
-        tabContents[tab] = createTabElement(this.win, this.storyUrl_, tab);
-        return tabContents;
-      },
-      {}
+    const container = this.element.querySelector(
+      '.i-amphtml-story-dev-tools-container'
+    );
+    this.tabContents_[DevToolsTab.PREVIEW] = createTabElement(
+      this.win,
+      this.storyUrl_,
+      DevToolsTab.PREVIEW
     );
+    this.tabContents_[DevToolsTab.PAGE_EXPERIENCE] = createTabElement(
+      this.win,
+      this.storyUrl_,
+      DevToolsTab.PAGE_EXPERIENCE
+    );
+    this.tabContents_[DevToolsTab.LOGS] = createTabLogsElement(
+      this.win,
+      this.storyUrl_,
+      DevToolsTab.LOGS
+    );
+    Object.values(this.tabContents_).forEach((tabContent) => {
+      container.appendChild(tabContent);
+      toggle(tabContent, false);
```

Comment:
Can you please double check that the build/layoutCallback aren't called since you're hiding them after they're connected to the DOM? I think it should be fine but might be worth inverting both lines for clarity.

Reply:
They are called anyways, no matter if you toggle them before appending or after. Assuming the other PR gets merged, we can move this convo to there

Reply:
`layoutCallback()` should not be called if the element is hidden, this is enforced by AMP. We should find out why it doesn't work and fix it, so we don't validate the story in the background when we're on the preview tab.